scripts/export_environment.py

```diff
@@ -170,32 +160,45 @@ def resolve_env(env: Environment) -> Environment:
     >>> resolve_env({'x': '{y}', 'y': '42'})
     {'x': '42', 'y': '42'}
 
-    Literal curly braces need to be escaped:
+    Transitive references are supported:
 
-    >>> resolve_env({'x': '{{', 'y': '}}', 'z': '{{x}}'})
-    {'x': '{', 'y': '}', 'z': '{x}'}
+    >>> resolve_env({'x': '{y}', 'y': '{z}', 'z': '42'})
+    {'x': '42', 'y': '42', 'z': '42'}
 
-    Does not support transitive references:
+    Generative references formed during resolution are allowed provided that
+    curly braces are escaped:
 
-    >>> resolve_env({'x': '{y}', 'y': '{z}', 'z': '42'})
-    {'x': '{z}', 'y': '42', 'z': '42'}
+    >>> resolve_env({'x': '{{{y}}}', 'y': 'z', 'z': '42'})
+    {'x': '42', 'y': 'z', 'z': '42'}
 
-    Transitive resolution would yield
+    Compared to:
 
-    {'x': '42', 'y': '42', 'z': '42'}
+    >>> resolve_env({'x': '{{y}}', 'y': 'z', 'z': '42'})
+    {'x': 'z', 'y': 'z', 'z': '42'}
 
-    Does not support generative ones (references formed during resolution):
+    References to a missing key yields an empty string replacement:
 
-    >>> resolve_env({'x': '{y}}}', 'y': '{{z', 'z': '42'})
-    {'x': '{{z}', 'y': '{z', 'z': '42'}
+    >>> resolve_env({'x': 'y is {y}'})
+    {'x': 'y is '}
 
-    which could be
+    Unmatched curly braces are not allowed:
 
-    {'x': '42', 'y': '{z', 'z': '42'}
+    >>> resolve_env({'x': '{y'})
+    Traceback (most recent call last):
+    ...
+    ValueError: expected '}' before end of string
     """
-    # `{missing_var}` references in values should yield the empty string,
```

Comment:
Why did you think it necessary to remove this? I see you added a test but the motivation is lost.

Reply:
The comment was removed as it seemed repetitive now that the behavior is documented in the docstring test. It was an oversight on my part to assume the the motivation of the behavior wasn't necessary to document.